static/templates/single_message.handlebars

```diff
@@ -6,32 +6,32 @@
         <div class="messagebox-border">
             <div class="messagebox-content">
                 <div class="message_top_line">
-                    {{#include_sender}}
                     <span class="message_sender{{^status_message}} sender_info_hover{{/status_message}}">
-                        {{! See ../js/notifications.js for another user of avatar_url. }}
-                        <div class="u-{{msg/sender_id}} inline_profile_picture{{#status_message}} sender_info_hover{{/status_message}}"
-                             style="background-image: url('{{small_avatar_url}}');"/>
-                        {{#if status_message}}
-                            <span class="sender-status">
-                                <span class="sender_name-in-status sender_info_hover">{{msg/sender_full_name}}</span>
+                        {{#include_sender}}
+                            {{! See ../js/notifications.js for another user of avatar_url. }}
+                            <div class="u-{{msg/sender_id}} inline_profile_picture{{#status_message}} sender_info_hover{{/status_message}}"
+                                 style="background-image: url('{{small_avatar_url}}');"/>
+                            {{#if status_message}}
+                                <span class="sender-status">
+                                    <span class="sender_name-in-status sender_info_hover">{{msg/sender_full_name}}</span>
+                                    {{#if sender_is_bot}}
+                                    <img class="bot-icon" title="{{t 'Bot' }}" src="/static/images/bot.svg" />
+                                    {{/if}}
+                                    <span class="status-message">
+                                        {{{ status_message }}}
+                                    </span>
+                                    {{#if_and last_edit_timestr include_sender}}
+                                    <div class="message_edit_notice" title="{{#tr this}}Edited (__last_edit_timestr__){{/tr}}">{{t "EDITED" }}</div>
+                                    {{/if_and}}
+                                </span>
+                            {{else}}
+                                <span class="sender_name">{{msg/sender_full_name}}</span>
```

Comment:
Basically, the way the position of the user profile popover is calculated is by comparing it to the position of the `.sender_info_hover` span element. When no message author is displayed, `.sender_info_hover` isn't rendered. I changed the position of the `{{#include_sender}}` template tag so that `.sender_info_hover` will always render, but it will be null in the event that no message author is displayed.